hs/CalendarResponse.hs

```diff
@@ -0,0 +1,230 @@
+module CalendarResponse where
+
+import Data.List (sort)
+import Data.List.Split (splitOn)
+import Data.Time (Day, UTCTime, addDays, formatTime, getCurrentTime, utctDay)
+import Happstack.Server (ServerPart, Response, toResponse)
+import Control.Monad.IO.Class (liftIO)
+import System.Locale
+import Config (firstMondayFall, lastWednesdayFall ,firstMondayWinter, lastMondayWinter, outDay)
+import Database.CourseQueries (returnTutorialTimes, returnLectureTimes)
+import qualified Data.Text as T
+import Database.Tables as Tables
+
+-- | Returns a CSV file of events as requested by the user.
+calendarResponse :: String -> ServerPart Response
+calendarResponse courses =
+    liftIO $ getCalendar courses
+
+-- | Gets together all the pieces of the program.
+getCalendar :: String -> IO Response
+getCalendar courses = do
+    let courseInfo = getCoursesInfo courses
+    databaseInfo <- sequence $ map pullDatabase courseInfo
+    currentTime <- getCurrentTime
+    let timeSystem = getTime currentTime
+    let events = concat $ map (getEvent timeSystem) databaseInfo
+    return $ toResponse $ getCSV $ events
+
+-- | Generates a properly formatted current date and time.
+getTime :: UTCTime -> String
+getTime currentTime = date ++ "T" ++ hour ++ "Z"
+    where
+    date = formatTime defaultTimeLocale "%Y%m%d" (utctDay currentTime)
+    hour = formatTime defaultTimeLocale "%H%M%S" currentTime
+
+-- | Generates a string representing a CSV file.
+getCSV :: [String] -> String
+getCSV events =  if events == [] then "" else unlines $ header ++ events ++ bottom
+    where
+    header = ["BEGIN:VCALENDAR", "VERSION:2.0", "PRODID:-//Courseography//Calendar", "CALSCALE:GREGORIAN", "METHOD:PUBLISH"]
```

Comment:
Thank you for all your comments. I will change the format for both the where statements and the one liners.